### PR TITLE
Use `msys2` shell on Windows GHA runners

### PR DESCRIPTION
**Pull request by jasagredo**

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]

**Review by Kleidukos — APPROVED** [review]

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -60,28 +54,33 @@ env:
...
+          - { os: ubuntu-latest,  shell: bash }
+          - { os: macos-13,       shell: bash}
         # If you remove something from here, then add it to the old-ghcs job.
         # Also a removed GHC from here means that we are actually dropping
         # support, so the PR *must* have a changelog entry.
         ghc: ['9.10.1', '9.8.2', '9.6.4', '9.4.8', '9.2.8', '9.0.2', '8.10.7', '8.8.4', '8.6.5']
         exclude:
           # corrupts GHA cache or the fabric of reality itself, see https://github.com/haskell/cabal/issues/8356
-          - os: windows-latest
+          - sys: { os: windows-latest, shell: 'C:/msys64/usr/bin/bash.exe -e {0}' }
```
Comment (ulysses4ever) [inline .github/workflows/validate.yml:73]: A little sad that you have to repeat this shell incantation in every include/exclude clause. Imagine this shell call will change a little... Are you positive that leaving out the `shell` field here is not acceptable?

**Review by ulysses4ever — APPROVED** [review]
Makes sense, thank you.

File: .github/workflows/validate.yml (hunk 1 above)
Comment (jasagredo) [reply .github/workflows/validate.yml:73]: I tried to exclude by:
```
exclude:
  - sys.os: windows-latest
  - ghc: '8.10.7'
```
But it doesn't work :(

Comment (ulysses4ever) [reply .github/workflows/validate.yml:73]: Oh well, that's okay then. (I imagined that you did try it, and just wanted to double check.)

One day their workflow description syntax will support global constants...

**Comment by geekosaur** [comment]
We normally use Mergify for this, rather than merging directly.

**Comment by Kleidukos** [comment]
I vouch for this direct merge, I didn't see the point in going through applying two labels (merge and delay-passed) for this.

**Comment by ulysses4ever** [comment]
I'd prefer that we stick to Mergify unless there's a feeling of urgency.

**Pull request merged**